client/src/components/App.js

```diff
@@ -26,11 +27,19 @@ const App = (props) => {
     fetchCurrentUser()
   }, [])
 
+  let newPostComponent
+  if (currentUser) {
+    newPostComponent = NewSaladForm
+  } else {
+    newPostComponent = RegistrationForm
+  }
+
```

Comment:
nope, unfortunately Im not going to allow this. You need to use the `AuthenticatedRoute` pattern shown in the user auth clinic and readings